### ISSUE TITLE
COLLECT/INTO adds to the tail of the COLLECTED block

### ISSUE THREAD

**Issue by rgchris** [red/red#2927]
Would expect to insert new values at the index of the given block:

```
>> also target: [] repeat x 3 [collect/into [keep x] target]
== [1 2 3]
```

Expected:

```
>> also target: [] repeat x 3 [collect/into [keep x] target]
== [3 2 1]
```

**Comment by greggirwin** [comment]
You can see in the source of `collect` that `append` is used. Different than R3. The `/into` doc string does say "insert", so maybe that should be changed.

**Comment by greggirwin** [comment]
Different use cases. Insert does allow more control from the outside.

**Comment by dockimbel** [comment]
Relates to #2189.

Changing the `/into` behavior would need to be done everywhere at the same time.

**Comment by rgchris** [comment]
@dockimbel My bad, I did a search for 'collect into' before posting, didn't see that in the results...

**Comment by hiiamboris** [comment]
I vote for appending behavior, for it being the most common case and not letting you shoot yourself into the foot performance-wise.

**Comment by greggirwin** [comment]
How is this for a new `/into` doc string?

"Append to a buffer instead of creating a new block"

**Comment by greggirwin** [comment]
Or "Put results in out block, instead of creating a new block."

Here `out` is the param name used in `load/into`. We should sync up the param name and doc strings for all `/into` cases if possible.  `Transcode` uses `dst`, `output` is clearer than `out`, maybe `output`? `Out` is not a normal noun, and doesn't read well to me, e.g.:

```
    out: case [
        part [transcode/part source length] 
        next [
            set position second out: transcode/next source 
            return either :all [reduce [out/1]] [out/1]
        ] 
        'else [transcode source]
    ] 
    either trap [out] [
        unless :all [if 1 = length? out [out: out/1]] 
        out
    ]
```

Providing a target buffer is a pattern used in other cases as well, it would be fantastic to find a good name we can add to recommendations in the style guide. I don't like `buf/buffer` and `dest` as much where it's used for a return value. For a modified arg, optional or not, that is changed as a side effect, those are OK, but more precise names may be even better. e.g. a function that logs actions taken in addition to a result.

`Output`  as the `/into` param name?

**Comment by greggirwin** [comment]
https://github.com/red/red/wiki/%5BDOC%5D-Coding-Style-Guide#common-refinement-names section added. In just quick searches, it turned up a number of cases where names are not consistent. Pretty good overall, but worth noting. Something @meijeru might already have in a list somewhere.

**Comment by rebolek** [comment]
> I vote for appending behavior, for it being the most common case and not letting you shoot yourself into the foot performance-wise. When insertion is required, `insert <here> collect` will do it in a single operation.

I disagree as it goes against the purpose of `into`. `insert <here> collect` needs creation of a new buffer before inserting it on desired position. `append` behaviour can be easily emulated with `insert` but not vice-versa.

**Comment by hiiamboris** [comment]
@rebolek I suggest reading the discussion in https://github.com/red/red/issues/3340 to understand my vote deeper. But TL;DR: don't assume, benchmark :)
